src/background/Wallet/Wallet.ts

```diff
@@ -1133,59 +1140,82 @@ export class Wallet {
 
   async addEthereumChain({
     context,
-    params: { values, origin },
+    params: { values, origin, chain: chainStr, created },
   }: WalletMethodParams<{
-    values: [NetworkConfig];
+    values: [AddEthereumChainParameter];
     origin: string;
+    chain?: string;
+    created?: string;
   }>) {
     this.verifyInternalOrigin(context);
+    const chain = chainStr || toCustomNetworkId(values[0].chainId);
+    // NOTE: This is where we might want to call something like
+    // {await networksStore.loadNetworkConfigByChainId(values[0].chainId)}
+    // IF we wanted to refactor networkStore to not hold searched values
     const result = chainConfigStore.addEthereumChain(values[0], {
+      id: chain,
       origin,
+      created: created != null ? Number(created) : undefined,
     });
 
-    this.emitter.emit('chainChanged', createChain(values[0].chain), origin);
+    this.emitter.emit('chainChanged', createChain(chain), origin);
     emitter.emit('addEthereumChain', {
       values: [result.value],
       origin: result.origin,
     });
     return result;
   }
 
-  async removeEthereumChain({
+  async resetEthereumChain({
     context,
     params: { chain: chainStr },
   }: WalletMethodParams<{ chain: string }>) {
     this.verifyInternalOrigin(context);
     this.ensureRecord(this.record);
-    const chain = createChain(chainStr);
-    chainConfigStore.removeEthereumChain(chain);
-    // We need to update all permissions which point to the removed chain
+    chainConfigStore.removeEthereumChain(createChain(chainStr));
+    this.verifyOverviewChain();
+  }
+
+  async updateChainForAffectedOrigins({
```

Comment:
This doesn't need to be async
(But keep in mind that making it sync will make it unavailable to outside scripts, so see https://github.com/zeriontech/zerion-wallet-extension/pull/432/files#r1586302220

Reply:
bump